tests: adding test cases for IWDG on disco stm32f3 board

FRASTM opened the pull request:
This patch enables the IWDG so that the testcase **tests/drivers/watchdog/wdt_basic_api**
can run on this **stm32f3_disco** board from STMicroelectronics

Signed-off-by: Francois Ramu <francois.ramu@st.com>

ABOSTM reviewed (approved):
LGTM

erwango reviewed (changes requested):
In order to be run by sanitycheck script, you need to add '  - watchdog' in stm32f3_disco yaml supported section

FRASTM commented:
new commit added with modified stm32f3_disco.yaml

erwango reviewed (changes requested):
Having 2 commits in this case seems a bit overkill, please squash.
Also, please update the board documentation.

FRASTM committed:
boards: arm: stm32f3 disco board enables the iwdg

This patch enables the Independent watchdog (iwdg)
so that the testcase tests/drivers/watchdog/wdt_basic_api
can run on this stm32f3_disco board

Signed-off-by: Francois Ramu <francois.ramu@st.com>

FRASTM commented:
now squashed in a single commit

erwango commented on boards/arm/stm32f3_disco/doc/index.rst:
```diff
@@ -65,6 +65,7 @@ STM32F3DISCOVERY Discovery kit provides the following hardware components:
 - USB 2.0 full speed interface
 - Infrared transmitter
 - DMA Controller
+- independent watchdog
```
I was rather referring to the "Supported features" section. Please update

FRASTM replied:
ok, moved
